internal/services/cragcast/cragcast.go

```diff
@@ -0,0 +1,227 @@
+// this will contain functions for different forecast functionality.
+// eg. GetForecast, GetHumidity, GetRainfall, etc...(not necessarily these, just giving them as examples to illustrate)
+
+package cragcast
+
+import (
+	"errors"
+	"sort"
+	"strings"
+	"time"
+
+	client "github.com/dsauerbrun/cragcast/pkg/weather-client"
+	noaaTypes "github.com/dsauerbrun/cragcast/pkg/weather-client/noaa"
+)
+
+func GetForecast(cragId int) (*Cragcast, error) {
+	// TODO: get crag's lat/lng from DB based on cragId. For now, we use boulder which is 40.0294122,-105.3223779
+	lat, lng := 40.0294122, -105.3223779
+	weatherClientForecast, err := client.New().GetForecast(lat, lng)
```

Comment:
Every invocation of `GetForcast` should not generate a new client.

Thus, If we keep this new package cragcast (see [comment here](https://github.com/dsauerbrun/cragcast/pull/9/files#r1181073623)), we'll want to introduce a `CragCast` struct where the weather-client is a member. `GetForcast` would then become a method on it.

Reply:
Waiting on discussion from https://github.com/dsauerbrun/cragcast/pull/9/files#r1181073623 to be resolved before doing this work